### PR TITLE
Feature/Replace options' svg with inline background-image

### LINKED ISSUE #304: [🔨for options]: change the way we use icons
As a UIplayground consumer, I get the problem that I can't add icons during integration. You can see it on the screenshot below.

image

**Possible Solution** is to move the icons to HTML or CSS files

**Note**: The issue https://github.com/exadel-inc/ui-playground/issues/306 can be fixed under this one.

### LINKED ISSUE #306: [🐛for options]: change the thickness of the theme icon
Please don't highlight the dark theme moon icon. We don't have the sun highlighted when we have the dark theme, so you don't need to highlight the moon.

### PR DESCRIPTION
**Pull request by Sisha3342**
closes #304 
closes #306

### CODE REVIEW

File: src/plugins/options/option/option.ts [hunk 1]
```diff
@@ -6,72 +6,60 @@ import {ENTER} from '@exadel/esl/modules/esl-utils/dom/keys';
...
+  }
 
-    @bind
-    protected _onKeydown(e: KeyboardEvent) {
-        if (ENTER !== e.key) return;
-        this.toggleState();
-        EventUtils.dispatch(this, 'uip:optionclick');
-    }
+  protected unbindEvents() {
+    this.removeEventListener('click', this._onClick);
```
Comment (yadamskaya) [inline src/plugins/options/option/option.ts:58]:
```suggestion
  protected unbindEvents() {
    this.removeEventListener('click', this._onClick);
    this.removeEventListener('keydown', this._onKeydown);
```

File: src/plugins/options/option/option.less [hunk 2]
```diff
@@ -1,47 +1,37 @@
...
...2R5IKdwBwEfEnTmPqnAHgLNKESZJUK88BfkH8idSYOmUtcD5pHQRJqrWDSUvpRp9YjSk5wxf+fZGkBpkCHAlcQfyJ1piSsho4k7QmgiQ12n7A6aSXm6JPvsZEZSXpOHgyktRnnkSaR2A58SdjY3qVpaT9fhukCjkVsOpoS+A44G3AFsG1SLncTlqe9yvAsuBaJKkoM4FjgT8Q/yvNmCqyFrgQOBoXWJKkSU0FXgycw+Or1xlTpywhLZq1C1KP+AhATbM18Drg9cDOsaVIk7oWOBU4g/RuiySpAvsBX8KXBk1ZWULaL5+GJCmrTYA3Ab8j/uRv+jMDpGf7xwKzkQrgIwD1mwOAVwFHAU8IrkXNdwPwLeDrwF3BtUjrcACgfjUVOJD0tvXR+I21qjOftNz110jP+KUiOQCQ1h0MvArYKrYc1dBdwLnAd4BLgmuRWuIAQFrXdOCvSAOBl+JEQxrftaRf+ufjL33VkAMAaXxTSSutHTyUg4D1QitSpAHgctKv/HPwmb5qzgGA1LotgBeSJh06lDQlsZrtNuCioVwIPBxbjlQdBwBSZ6aSvuN+EWlAsD/p8YHq7QHg5zx+0Z8fWo2UkQMAqRobAM8EnkN6ofAAYKPQitSK+4BLh/JL4DrSfPxS4zkAkPKYBuxGmo3w2aSBwR6hFQnSCnuXkt7Uv5THF5KS+o4DAKl3tiMNBJ4O7D0UPznMYwC4BfjtiFwDLI0sSiqJAwAp1pY8PhjYa+jPPYFZkUXVzBLgJtKv+eGL/fW4uI40IQcAUnmmkZaF3Zv02OCJwDxgJ2Bb0guI/WYAuIf0q/6modw89Of9gXVJteUAQKqXGcCOPD4g2Gnofw9nK+p5XC8jvZB3F3DnUOaP+N93A6ujipOaqI4nCknjmwJsTpqzYPNRmTvG/20m6WuF6aRVE7uxmnQhXzL05yNDfz5Meva+jPSZ3X1Dfz4ALAAWAiu63LakNjkAkDTSesCcCf4Z0gV+2Cp81i5JkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJkiRJklSW/w9w41bKAr1zIAAAAABJRU5ErkJggg=='/></defs></svg>"),
```
Comment (ala-n) [inline src/plugins/options/option/option.less:29]: This one is not vector, consider to optimize it

File: src/plugins/options/option/option.less [hunk 3]
```diff
@@ -1,47 +1,37 @@
...
-            vertical-align: middle;
-            fill: currentColor;
-            stroke: currentColor;
-        }
+    &.editor-collapsed-option {
+      width: 2rem;
     }
 }
+
+@option-icons: {
```
Comment (ala-n) [inline src/plugins/options/option/option.less:28]: Basically almost all of the icons are not optimized, you can save a lot of KB to users if you optimes the assets.

File: src/plugins/options/option/option.ts [hunk 4]
```diff
@@ -6,72 +6,60 @@ import {ENTER} from '@exadel/esl/modules/esl-utils/dom/keys';
...
 
-    protected connectedCallback() {
-        super.connectedCallback();
-        this.classList.add(`${this.attribute}-option`);
-        this.tabIndex = 0;
-        this.bindEvents();
-        this.render();
-    }
+  protected bindEvents() {
+    this.addEventListener('click', this._onClick);
```
Comment (ala-n) [inline src/plugins/options/option/option.ts:35]: Just a note:
You can cinsider upgrade ESL to 4th beta, then you can simplify a lot all listener managemant
CC: @yadamskaya

File: src/plugins/options/option/option.ts [hunk 5]
```diff
@@ -6,72 +6,60 @@ import {ENTER} from '@exadel/esl/modules/esl-utils/dom/keys';
...
+  }
 
-    protected bindEvents() {
-        this.addEventListener('click', this._onClick);
-        this.addEventListener('keydown', this._onKeydown);
-    }
+  @bind
+  protected _onClick() {
+    this.toggleState();
+    EventUtils.dispatch(this, 'uip:optionclick');
```
Comment (ala-n) [inline src/plugins/options/option/option.ts:42]: Note: Just $$fire in latest ESL beta

File: src/plugins/options/option/option.ts [hunk 6]
```diff
@@ -6,72 +6,60 @@ import {ENTER} from '@exadel/esl/modules/esl-utils/dom/keys';
...
-        this.removeEventListener('click', this._onClick);
-    }
-
-    public toggleState(force?: boolean) {
-        this.active = force === undefined ? !this.active : force;
-        this.classList.toggle('active', this.active);
-    }
+  public toggleState(force?: boolean) {
+    this.active = force === undefined ? !this.active : force;
+    this.classList.toggle('active', this.active);
```
Comment (ala-n) [inline src/plugins/options/option/option.ts:63]: this.$$cls('active', this.active);

**Review by nattallius — APPROVED** [review]

File: src/plugins/header/options/option/option.ts [hunk 7]
```diff
@@ -1,80 +1,66 @@
...
+    this.toggleState();
+    this.$$fire('uip:optionclick');
+  }
 
-    protected render() {
-        this.icon && this.append(this.icon);
-    }
+  @bind
+  protected _onKeydown(e: KeyboardEvent) {
+    if (ENTER !== e.key) return;
```
Comment (ala-n) [inline src/plugins/header/options/option/option.ts:47]: There is also an a11y behavior that the actionable elements behaves also on the SPACE key and process it usually as an click, ao in our scenario it is the same as ENTER

File: src/plugins/header/options/option/option.ts [hunk 8]
```diff
@@ -1,80 +1,66 @@
...
+    this.toggleState();
+    this.$$fire('uip:optionclick');
```
Comment (ala-n) [inline src/plugins/header/options/option/option.ts:49]:
```suggestion
    this.$$fire('uip:option:checked');
```
```suggestion
    this.$$fire('uip:option:activated');
```
```suggestion
    this.$$fire('uip:option:changed');
```

**Review by ala-n — APPROVED** [review]

**Review by PolinaVolchetskaya — APPROVED** [review]

**Review by AlexanderBazukevich — APPROVED** [review]

**Pull request merged**